File: MsaCookingApp.Contracts/Features/Meals/DTOs/SpoonacularGetMealInstructions.cs
namespace MsaCookingApp.Contracts.Features.Meals.DTOs;

public class SpoonacularGetMealInstructions
{
    public IEnumerable<SpoonacularGetMealInstructionStepDto> Steps { get; set; } =
        new List<SpoonacularGetMealInstructionStepDto>();
}

File: msa_cooking_app_client/lib/features/meals/models/spoonacular_get_meal_instructions.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:msa_cooking_app_client/features/meals/models/spoonacular_get_meal_instruction_step.dart';

part 'spoonacular_get_meal_instructions.g.dart';

@JsonSerializable()
class SpoonacularGetMealInstructions {
  final List<SpoonacularGetMealInstructionStep> steps;

  SpoonacularGetMealInstructions(this.steps);

  factory SpoonacularGetMealInstructions.fromJson(Map<String, dynamic> json) =>
      _$SpoonacularGetMealInstructionsFromJson(json);

  Map<String, dynamic> toJson() => _$SpoonacularGetMealInstructionsToJson(this);
}